src/braket/default_simulator/openqasm/circuit.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from typing import List, Optional
+
+from braket.ir.jaqcd.program_v1 import Results
+from braket.ir.jaqcd.shared_models import Observable, OptionalMultiTarget
+
+from braket.default_simulator.operation import GateOperation, KrausOperation
+from braket.default_simulator.operation_helpers import from_braket_instruction
+from braket.default_simulator.result_types import _from_braket_observable
+
+
+class Circuit:
+    """
+    This is a lightweight analog to braket.ir.jaqcd.program_v1.Program.
+    The Interpreter compiles to an IR to hand off to the simulator,
+    braket.default_simulator.state_vector_simulator.StateVectorSimulator, for example.
+    Our simulator module takes in a circuit specification that satisfies the interface
+    implemented by this class.
+    """
+
+    def __init__(
+        self,
+        instructions: Optional[List[GateOperation]] = None,
+        results: Optional[List[Results]] = None,
+    ):
+        self.instructions = []
+        self.results = []
+        self.qubit_set = set()
+
+        if instructions:
+            for instruction in instructions:
+                self.add_instruction(instruction)
+
+        if results:
+            for result in results:
+                self.add_result(result)
+
+    def add_instruction(self, instruction: [GateOperation, KrausOperation]) -> None:
+        """
+        Add instruction to the circuit.
+
+        Args:
+            instruction (GateOperation): Instruction to add.
+        """
+        self.instructions.append(instruction)
+        self.qubit_set |= set(instruction.targets)
```

Comment:
nit:
```suggestion
        self.qubit_set.update(instruction.targets)
```
to remove that extra `set()` cast